### PR TITLE
Removed needless login shell from chroot calls where possible

### PR DESCRIPTION
**Pull request by jsmeix**
see https://github.com/rear/rear/issues/862

### CODE REVIEW

**Comment by jsmeix** [comment]
The only "chroot ... bash --login" call that is left is in
finalize/Debian/i386/170_rebuild_initramfs.sh
<pre>
  chroot $TARGET_FS_ROOT /bin/bash --login -c "/usr/share/mdadm/mkconf >/etc/mdadm/mdadm.conf"
</pre>
because of the stdout redirection.

**Comment by jsmeix** [comment]
Do not merge it!
It does not yet work.
This one is RFC 1925 (8): "It is more complicated than you think."

**Comment by jsmeix** [comment]
Now it works for me on SLES12:
<pre>
RESCUE e205:~ # rear -d -D recover
...
Restoring finished.
...
Running mkinitrd...
Recreated initrd (/sbin/mkinitrd).
Installing GRUB2 boot loader
Finished recovering your system. You can explore it under '/mnt/local'.
</pre>

FYI:
I need in local.conf
<pre>
OS_MASTER_VENDOR="SUSE_LINUX/i386"
</pre>
(or something else like that) because otherwise
finalize/SUSE_LINUX/i386/170_rebuild_initramfs.sh
is not sourced.
Whether or not a rebuild_initramfs script should be sourced
in any case is a different issue - I never had issues with the
initrd on my test systems so that a rebuild_initramfs script
seems to be not really mandatory.

**Comment by jsmeix** [comment]
@gdha 
could you please review whether or not
it still works for you with my changes in
finalize/Fedora/i386/170_rebuild_initramfs.sh

FYI:
By the way I found a chroot call in
finalize/Linux-i386/220_install_grub2.sh
where I think that cannot work:
<pre>
chroot $TARGET_FS_ROOT $grub_name-mkconfig -o /boot/$grub_name/grub.cfg
</pre>
because $grub_name evaluates to "grub2" or "grub"
but chroot requires usually a full path to be executed like
<pre>
RESCUE e205:~ # chroot /mnt/local ifconfig
chroot: failed to run command 'ifconfig': No such file or directory

RESCUE e205:~ # chroot /mnt/local /sbin/ifconfig
eth0      Link encap:Ethernet  HWaddr 52:54:00:25:A4:49
...
</pre>
see also my new code regarding "mkinitrd_binary" in
finalize/Fedora/i386/170_rebuild_initramfs.sh

**Comment by gdha** [comment]
@jsmeix Merged it - results will be posted in https://github.com/gdha/rear-automated-testing/issues/13

**Comment by jsmeix** [comment]
@gdha 
many thanks for your careful review!

FYI regarding who the 'I' actually is:
To find out who made what in a file I use usually
the git command "git log -p --follow file.sh".
For the 170_rebuild_initramfs.sh files
that command shows as the oldest entry always
<pre>
commit 844d50b75ac4b7722f4fee7a5ee3350b93f3adb7
Author: Schlomo Schapiro <github@schlomo.schapiro.org>
Date:   Sun Jun 6 08:30:21 2010 +0000

    - Integrated P2V patch from Heinlein Support. We start with 1.9 now to
</pre>
It is the same commit
https://github.com/rear/rear/commit/844d50b75ac4b7722f4fee7a5ee3350b93f3adb7
for
finalize/Fedora/i386/170_rebuild_initramfs.sh
finalize/Debian/i386/170_rebuild_initramfs.sh
finalize/SUSE_LINUX/i386/170_rebuild_initramfs.sh
so that I think the 'I' could be Peer Heinlein
and Schlomo only comitted his changes?

**Comment by gdha** [comment]
@jsmeix I can confirm the modifications work well on CentOS7

File: usr/share/rear/finalize/Debian/i386/170_rebuild_initramfs.sh [hunk 1]
```diff
@@ -1,7 +1,10 @@
 # rebuild the initramfs if the drivers changed
 #
-# probably not required, but I prefer to rely on this information when it
-# is backed by udev
+# probably not required, but I prefer to rely on this information when it is backed by udev
+# FIXME: who is 'I'?
+# Perhaps Schlomo Schapiro or someone who made the "P2V patch from Heinlein Support"?
```
Comment (schlomo) [inline usr/share/rear/finalize/Debian/i386/170_rebuild_initramfs.sh]: This is really long time ago. udev back then was indeed optional on some distros. I would say that this changed and udev is not optional any more.

**Comment by jsmeix** [comment]
@schlomo 
many thanks for the information!

I will do a separated submitrequest
where I let ReaR show same kind of warning
as near the end of the code about 
"Cannot create initramfs"
if the udev requirement is not fulfilled.

Regarding LogPrint messages like
<pre>
WARNING:
Cannot create initramfs
(found no update-initramfs in the recreated system).
Check the recreated system (mounted at /mnt/local)
and decide yourself, whether the system will boot or not.
</pre>

@schlomo
because I know what you think about warning messages:
Schould I remove the word "WARNING:" from those
messages and make it a plain information?

**Comment by schlomo** [comment]
Thanks @jsmeix for your consideration.

One one hand the warning is justified as it really means "I don't know".

On the other hand we should try to give the user even more information here. For example, which systems don't have udev nowadays? Maybe we can say that ReaR 2 actually requires udev and fails without? Then you could get rid of this warning.

Another thought: Maybe we should expand this to be a generic feature and collect such warnings in a list of  `RECOVERY_NOTICE[@]` which we display at the end. In an automated recovery having a non-empty list of recovery notices would be an error, in a manual recovery it would be printed as user info.

**Comment by jsmeix** [comment]
I have to think a bit more about it.
I think the idea behind something like a RECOVERY_NOTICE array
is related to https://github.com/rear/rear/issues/1134
because also in an automated recovery proper exit codes
could be used to signal something to the user.
In other words:
If in an automated recovery a RECOVERY_NOTICE
would be an error, rear should exit with an appropriate
non-zero exit code in this case to singal the user:
Not a fatal error (e.g. after calling the Error function)
but also not a clean recovery.

**Comment by jsmeix** [comment]
It seems using no login shell for chroot calls
causes more issues that it solves, see
https://github.com/rear/rear/pull/1345

**Pull request merged**